src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,59 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents one of a person's contact info in the address book
+ * Guarantees: immutable; is valid as declared in {@link #isValidContact(String, String)}
+ * @author brea
+ *
+ */
+public class Contact {
+		
+	public final String value;
+	protected boolean isPrivate;
+	
+    /**
+     * Validates given contact information using regex provided
+     *
+     * @throws IllegalValueException if given contact string is invalid
+     */
+	public Contact(String contact, boolean isPrivate, String validationRegex, String messageContactConstraint) throws IllegalValueException {
```

Comment:
Interesting design by capturing the validation regex and messages in constructor.